ansible_base/rbac/validators.py

```diff
@@ -133,6 +132,17 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
                         {'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include change, got: {display_perms}'}
                     )
 
+    if (not managed) and (not settings.ALLOW_SHARED_RESOURCE_CUSTOM_ROLES):
+        for perm in permissions:
+            # View permission for shared objects is interpreted as permission to view
+            # the resource locally, which is needed to be able to view parent objects
+            # For system roles, this exception is not allowed
+            if content_type and perm.codename.startswith('view'):
+                continue
+            model = perm.content_type.model_class()
+            if permission_registry.get_resource_prefix(model) == 'shared':
+                raise ValidationError({'permissions', 'Local custom roles can only include view permission for shared models'})
```

Comment:
We shouldn't raise this error if the local server is the provider for that shared resource.

Reply:
On second thought, no. That's what `ALLOW_SHARED_RESOURCE_CUSTOM_ROLES` is for, even if it's redundant with provider status.